components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1232,6 +1232,33 @@ where
             "to_peer_id" => msg.get_to_peer().get_id(),
         );
 
+        let msg_type = msg.get_message().get_msg_type();
+        let store_id = self.ctx.store_id();
+
+        if disk::disk_full_precheck(store_id) || disk::is_disk_full() {
+            let mut flag = false;
+            if MessageType::MsgAppend == msg_type {
+                let entries = msg.get_message().get_entries();
+                for i in entries {
+                    let entry_type = i.get_entry_type();
+                    if EntryType::EntryNormal == entry_type && !i.get_data().is_empty() {
```

Comment:
`MsgAppend` may contain multiple entries, I think we should not drop the message if there are `confchange` entries. /cc @hicqu

Reply:
So if it contains lots of large normal entries, TiKV can still run out of disk space.

I think both 2 implementations are reasonable. We can make the decision later in another PR after the design document is posted and confirmed.